lib/cli.js

```diff
@@ -21,82 +21,70 @@ var fs = require('fs');
 var getGitFolder = require('./getGitFolder');
 var validateMessage = require('../index');
 
-// hacky start if not run by mocha :-D
-// istanbul ignore next
-if (process.argv.join('').indexOf('mocha') === -1) {
-  var bufferToString = function (buffer) {
-    var hasToString = buffer && typeof buffer.toString === 'function';
-    return hasToString && buffer.toString();
-  };
+var commitMsgFileOrText = process.argv[2];
+var commitErrorLogPath = process.argv[3];
 
-  var getFileContent = function (filePath) {
-    try {
-      var buffer = fs.readFileSync(filePath);
-      return bufferToString(buffer);
-    } catch (err) {
-      // Ignore these error types because is most likely it is validating
-      // a commit from a text instead of a file
-      if(err && err.code !== 'ENOENT' && err.code !== 'ENAMETOOLONG') {
-        throw err;
-      }
-    }
-  };
+// On running the validation over a text instead of git files such as COMMIT_EDITMSG and GITGUI_EDITMSG
+// is possible to be doing that the from anywhere. Therefore the git directory might not be available.
+var gitDirectory;
+try {
+  gitDirectory = getGitFolder();
+
+  if (!commitErrorLogPath) {
+    commitErrorLogPath = gitDirectory + '/logs/incorrect-commit-msgs';
+  }
+} catch (err) {}
+
+validate(commitMsgFileOrText);
+
+
+function bufferToString(buffer) {
```

Comment:
I tried to find what could happen.

The guard was:
```js
if (process.argv.join('').indexOf('mocha') === -1) {}
```

So I added at the beginning of the file:
```js
console.warn('process.argv:', process.argv)
```
and started `npm run test` - result:
```
      process.argv: [ '/usr/bin/nodejs',
        '/home/cedric/Projects/validate-commit-msg/lib/cli.js' ]
```
Can't reproduce this case, looks like it was always going inside this `if`.
Do you have some clues to reproduce ? @kentcdodds ?